src/charm.py

```diff
@@ -329,43 +329,50 @@ def _get_current_hash(self) -> str:
             logger.debug("Unrecognized hash file format: %s", contents[:100])
             return self._hash_placeholder
 
-    @property
-    def _stored_hash(self) -> Optional[str]:
+    def _stored_get(self, key: str) -> Optional[str]:
         if relation := self.model.get_relation(self._peer_relation_name):
-            return relation.data[self.app].get("hash", None)
+            return relation.data[self.app].get(key, None)
         return None
 
-    @_stored_hash.setter
-    def _stored_hash(self, sha: str):
+    def _stored_set(self, key: str, value: str):
         """Update peer relation data with the given hash."""
         if not self.unit.is_leader():
-            logger.info("store hash: abort: not leader")
+            logger.info("store %s: abort: not leader", key)
```

Comment:
Aren't we spamming too much in `.info` level?
Is `.debug` a better level for this kind of messages?

Reply:
Good point! Fixed in next commit.